src/Features/CSharp/Portable/BraceCompletion/InterpolationBraceCompletionService.cs

```diff
@@ -80,18 +89,8 @@ public static async Task<bool> IsPositionInInterpolationContextAsync(Document do
                 return false;
             }
 
-            // For the case: string s = $"{}$$
-            // Here the token to the left is a close brace token inside an interpolation.
-            if (token.IsKind(SyntaxKind.CloseBraceToken) && token.Parent.IsKind(SyntaxKind.Interpolation))
-            {
-                return true;
-            }
-
-            return token.IsKind(
-                SyntaxKind.InterpolatedStringStartToken,
-                SyntaxKind.InterpolatedVerbatimStringStartToken,
-                SyntaxKind.InterpolatedStringTextToken,
-                SyntaxKind.InterpolatedStringEndToken);
+            // We can be starting an interpolation expression if we're inside an interpolated string.
+            return token.GetAncestor((node) => node.IsKind(SyntaxKind.InterpolatedStringExpression)) != null;
```

Comment:
hrmm... this is looking up arbitrarily far.  i.e. you could be in block in a lambda in an interpolation.  Is that ok?

Reply:
Ah good point...
I'll add a test case for that and see if I can work out a better check.  The reason I went for ancestor was that the InterpolatedStringExpression node could be a couple nodes up from the current token, but I definitely need to limit it